### PR TITLE
[Parallel] Add serializable object passing to FileProcessors

### PR DESCRIPTION
**Pull request by TomasVotruba**
Ref https://github.com/rectorphp/rector/issues/2971

### CODE REVIEW

File: .github/workflows/build_scoped_rector.yaml [hunk 1]
```diff
@@ -28,7 +28,7 @@ jobs:
             -
                 name: "Wait 3 minutes before checkout rectorphp/rector-src on create a tag"
                 if: "startsWith(github.ref, 'refs/tags/')"
-                run: sleep 180
+                run: sleep 120
```
Comment (samsonasik) [inline .github/workflows/build_scoped_rector.yaml:31]: task name need to be updated to `Wait 2 minutes...`

Comment (TomasVotruba) [reply .github/workflows/build_scoped_rector.yaml:31]: Thanks

File: src/Application/ApplicationFileProcessor.php [hunk 2]
```diff
@@ -42,23 +60,38 @@ public function __construct(
      * @param File[] $files
      * @return array{system_errors: SystemError[], file_diffs: FileDiff[]}
      */
-    public function run(array $files, Configuration $configuration): array
+    public function run(array $files, Configuration $configuration, InputInterface $input): array
     {
+        // no files found
+        $filesCount = count($files);
+        if ($filesCount === 0) {
```
Comment (samsonasik) [inline src/Application/ApplicationFileProcessor.php]: can use `$files === []` for performance improvement:

```suggestion
        if ($files === []) {
```

**Comment by samsonasik** [comment]
The diff for no change seems should no need to be displayed (counted), ref https://github.com/rectorphp/rector-src/runs/4648198614?check_suite_focus=true#step:6:34

```diff


2) rules-tests/DependencyInjection/Rector/Class_/ActionInjectionToConstructorInjectionRector/xml/services.xml

    ---------- begin diff ----------

    ----------- end diff -----------

                                                                                
 [OK] 2 files have been changed by Rector                                       
```

**Comment by TomasVotruba** [comment]
> The diff for no change seems should no need to be displayed

I'll check it, thanks for spotting :+1:

**Comment by samsonasik** [comment]
It seems surplus new line now:

```bash

Run bin/rector process rules --ansi --no-progress-bar

                                                                                
 [OK] Rector is done!                                                           
```

which should be 1 line instead of 2 between Command and [OK] Rector is done! message.

**Comment by samsonasik** [comment]
@TomasVotruba it seems make build scoped error, ref https://github.com/rectorphp/rector-src/runs/4648294686?check_suite_focus=true#step:14:78

```bash
Parse error: rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/FlexForms/FlexFormsProcessor.php:32
    30|     {
    31|         if ([] === $this->flexFormRectors) {
  > 32|             return;
    33|         }
    34|         $domDocument = new \DOMDocument();
A function with return type must return a value in rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/FlexForms/FlexFormsProcessor.php on line 32
------------------------------------------------------------
Parse error: rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/Fluid/FluidFileProcessor.php:34
    32|     {
    33|         if ([] === $this->fluidRectors) {
  > 34|             return;
    35|         }
    36|         foreach ($this->fluidRectors as $fluidRector) {
A function with return type must return a value in rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/Fluid/FluidFileProcessor.php on line 34
------------------------------------------------------------
Parse error: rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php:41
    39|         // Prevent unnecessary processing
    40|         if ([] === $this->transformer) {
  > 41|             return;
    42|         }
    43|         $this->currentFileProvider->setFile($file);
A function with return type must return a value in rector-prefixed-downgraded/vendor/ssch/typo3-rector/src/FileProcessor/Yaml/Form/FormYamlFileProcessor.php on line 41
------------------------------------------------------------
Parse error: rector-prefixed-downgraded/vendor/rector/rector-nette/src/FileProcessor/NeonFileProcessor.php:44
    42|     {
    43|         if ($this->neonRectors === []) {
  > 44|             return;
    45|         }
    46|         $fileContent = $file->getFileContent();
A function with return type must return a value in rector-prefixed-downgraded/vendor/rector/rector-nette/src/FileProcessor/NeonFileProcessor.php on line 44
```

**Comment by TomasVotruba** [comment]
I see. That's due to change of `FileProcessorInterface`:

https://github.com/rectorphp/rector-src/pull/1583/files#diff-169c1ccc343b4c5f96ebdf832a9c74b5f61cd3209d4410bfd0739e7eb0bca132

But that's only in the docbocks, so there should not be any BC break :thinking: 

The FileProcessors in typo3 should be updated to return array of system errors and file diffs. That might help too

**Comment by TomasVotruba** [comment]
This might help https://github.com/sabbelasichon/typo3-rector/pull/2766

**Pull request merged**